File: SDKs/flat.io/2.13.0/typescript/src/sdk/models/operations/getscorerevision.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetScoreRevisionPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=revision" })
  revision: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=score" })
  score: string;
}


export class GetScoreRevisionQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=sharingKey" })
  sharingKey?: string;
}


export class GetScoreRevisionSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oAuth2: shared.SchemeOAuth2;
}


export class GetScoreRevisionRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetScoreRevisionPathParams;

  @SpeakeasyMetadata()
  queryParams: GetScoreRevisionQueryParams;

  @SpeakeasyMetadata()
  security: GetScoreRevisionSecurity;
}


export class GetScoreRevisionResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  flatErrorResponse?: shared.FlatErrorResponse;

  @SpeakeasyMetadata()
  scoreRevision?: shared.ScoreRevision;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: SDKs/flat.io/2.13.0/python/src/sdk/models/operations/getscorerevision.py
import dataclasses
from typing import Optional
from ..shared import security as shared_security
from ..shared import flaterrorresponse as shared_flaterrorresponse
from ..shared import scorerevision as shared_scorerevision


@dataclasses.dataclass
class GetScoreRevisionPathParams:
    revision: str = dataclasses.field(metadata={'path_param': { 'field_name': 'revision', 'style': 'simple', 'explode': False }})
    score: str = dataclasses.field(metadata={'path_param': { 'field_name': 'score', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class GetScoreRevisionQueryParams:
    sharing_key: Optional[str] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'sharingKey', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class GetScoreRevisionSecurity:
    o_auth2: shared_security.SchemeOAuth2 = dataclasses.field(metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    

@dataclasses.dataclass
class GetScoreRevisionRequest:
    path_params: GetScoreRevisionPathParams = dataclasses.field()
    query_params: GetScoreRevisionQueryParams = dataclasses.field()
    security: GetScoreRevisionSecurity = dataclasses.field()
    

@dataclasses.dataclass
class GetScoreRevisionResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    flat_error_response: Optional[shared_flaterrorresponse.FlatErrorResponse] = dataclasses.field(default=None)
    score_revision: Optional[shared_scorerevision.ScoreRevision] = dataclasses.field(default=None)
    
